lib/image_optim/cmd.rb

```diff
@@ -44,6 +86,38 @@ def check_status!
 
         fail SignalException, status.termsig
       end
+
+      def init_options!(args)
+        pgroup_opt = Gem.win_platform? ? :new_pgroup : :pgroup
+
+        if args.last.is_a?(Hash)
+          args.last[pgroup_opt] = true
+        else
+          args.push(pgroup_opt => true)
+        end
+      end
+
+      def cleanup_process(pid)
+        Thread.new do
+          Process.kill('-TERM', pid)
+          Process.detach(pid)
+          now = Time.now
+
+          while Time.now - now < 10
+            begin
+              Process.getpgid(pid)
```

Comment:
This is more idiomatically expressed as `Process.kill(0, pid)` (and also raises `ESRCH` when the process goes away).